### PR TITLE
main_v2 - Optimize Crater Lake - Fix Warnings

### PR DESCRIPTION
**Pull request by DelphineChiu**
Summary:
- Fix all warnings in CL layer and common layer.
- Refactor ME portion APIs

### CODE REVIEW

File: common/dev/adc.c [hunk 1]
```diff
@@ -74,19 +74,30 @@ bool adc_init()
 	return true;
 }
 
+int bat_3v_set_gpio(uint8_t sensor_num, void *arg)
+{
+	SET_GPIO_VALUE_CFG *cfg = arg;
```
Comment (GoldenBug) [inline common/dev/adc.c:79]: Check to make sure arg != NULL

File: common/dev/adc.c [hunk 2]
```diff
@@ -98,28 +109,18 @@ static bool adc_read_mv(uint8_t sensor_num, uint32_t index, uint32_t channel, in
...
-			gpio_set(A_P3V_BAT_SCALED_EN_R, GPIO_LOW);
-		} else {
-			val = val * sensor_config[snrcfg_sensor_num].arg0 /
-			      sensor_config[snrcfg_sensor_num].arg1;
-		}
-		*reading = (calculate_accurate_MBR(sensor_num, val) / 1000) & 0xFFFF;
-		sensor_config[snrcfg_sensor_num].cache = *reading;
-		sensor_config[snrcfg_sensor_num].cache_status = SNR_READ_ACUR_SUCCESS;
+		*reading = (float)val * cfg->arg0 / cfg->arg1 / 1000;
+		cfg->cache = *reading;
```
Comment (GoldenBug) [inline common/dev/adc.c:122]: check reading != NULL before use

File: common/dev/adm1278.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		} else if (sensor_num == SENSOR_NUM_PWR_HSCIN) {
+			// m = +6123 * Rsense(mohm), b = 0, R = -2
+			val = ((((msg.data[1] << 8) | msg.data[0]) * 100) / 1530.75);
+		}
+	} else {
+		cfg->cache_status = SENSOR_FAIL_TO_ACCESS;
+		printf("Sensor number %x read fail\n", sensor_num);
+		return false;
+	}
+	*reading = val;
```
Comment (GoldenBug) [inline common/dev/adm1278.c:76]: Check reading != NULL before use.

File: common/dev/guid.c [hunk 4]
```diff
@@ -20,8 +22,11 @@ uint8_t GUID_read(EEPROM_ENTRY *entry)
 		return GUID_OUT_OF_RANGE;
 	}
 
-	memcpy(&entry->config, &guid_config[entry->config.dev_id],
-	       sizeof(guid_config[entry->config.dev_id]));
+	if (entry->config.dev_id >= (sizeof(guid_config) / sizeof(EEPROM_CFG))) {
+		return GUID_OUT_OF_RANGE;
+	}
+
+	memcpy(&entry->config, &guid_config[entry->config.dev_id], sizeof(EEPROM_CFG));
```
Comment (GoldenBug) [inline common/dev/guid.c:29]: We should check that entry != NULL when passed in

File: common/dev/isl69260.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+#include "sensor.h"
+#include "hal_i2c.h"
+#include "plat_sensor_table.h"
+
+#define RETRY 5
+
+struct isl69260_page_data isl69260_page_data_args[] = { [0] = { 0x0, 0x0 }, [1] = { 0x0, 0x1 } };
+
+int isl69260_switch_page(uint8_t sensor_num, void *args)
```
Comment (GoldenBug) [inline common/dev/isl69260.c:13]: check args != NULL

File: common/dev/isl69260.c [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+	if (i2c_master_write(&msg, RETRY)) {
+		printf("sensor num %x switch page fail\n", sensor_num);
+		return -1;
+	}
+
+	return 0;
+}
+
+bool isl69260_read(uint8_t sensor_num, float *reading)
```
Comment (GoldenBug) [inline common/dev/isl69260.c:32]: check readings != NULL

File: common/dev/nvme.c [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+			cfg->cache_status = SENSOR_UNSPECIFIED_ERROR;
+			return false;
+		}
+		not_available_retry_num = 0;
+	} else {
+		cfg->cache_status = SENSOR_FAIL_TO_ACCESS;
+		return false;
+	}
+
+	*reading = (float)val;
```
Comment (GoldenBug) [inline common/dev/nvme.c:74]: check that reading != NULL

File: common/dev/pch.c [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		status = ipmb_read(bridge_msg, IPMB_inf_index_map[bridge_msg->InF_target]);
+		if (status != IPMB_ERROR_SUCCESS) {
+			printf("ipmb read fail status: %x\n", status);
+			cfg->cache_status = SENSOR_FAIL_TO_ACCESS;
+			free(bridge_msg);
+			return false;
+		}
+
+		if (bridge_msg->completion_code == CC_SUCCESS) {
+			*reading = convert_MBR_to_reading(sensor_num, bridge_msg->data[0]);
```
Comment (GoldenBug) [inline common/dev/pch.c:40]: Check that reading != NULL

File: common/dev/peci.c [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+				break;
+			default:
+				printf("PECI sensor [%x] control hardware, firmware or associated logic error\n",
+				       sensor_num);
+				cfg->cache_status = SENSOR_UNSPECIFIED_ERROR;
+				break;
+			}
+		}
+
+		free(read_buf);
```
Comment (GoldenBug) [inline common/dev/peci.c:69]: If we're going to use the SAFE_FREE macro we should use it everywhere.

File: common/dev/peci.c [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+
+#include "sensor.h"
+#include "hal_peci.h"
+#include "plat_sensor_table.h"
+
+#define PECI_READ_LEN 5
+#define PECI_RETRY_MAX 6
+
+#ifdef CONFIG_PECI
+static bool peci_get_power(uint8_t sensor_num, int *reading)
```
Comment (GoldenBug) [inline common/dev/peci.c:15]: Check reading != NULL

File: common/dev/peci.c [hunk 11]
```diff
@@ -0,0 +1,268 @@
...
+		if (ret == 0) {
+			if (*complete_code == PECI_CC_RSP_SUCCESS) {
+				break;
+			}
+		} else if (--retry_count) { // wait for 10ms before next retry
+			k_msleep(10);
+		} else { // all retries failed exit the loop
+			break;
+		}
+	}
```
Comment (GoldenBug) [inline common/dev/peci.c:44]: There seems to be a lot of duplicate looking code in this function. Might be worth looking and seeing if anything can be factored out into a helper function.

File: common/dev/peci.c [hunk 12]
```diff
@@ -0,0 +1,268 @@
...
+	last_run_time = run_time;
+
+	if (diff_time == 0) {
+		free(read_buf);
+		return false;
+	} else {
+		// energy / unit time
+		*reading = ((float)diff_energy / (float)diff_time * 0.06103515625);
+		free(read_buf);
```
Comment (GoldenBug) [inline common/dev/peci.c:154]: SAFE_FREE

File: common/dev/peci.c [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+		return true;
+	}
+}
+
+bool peci_sensor_read(uint8_t sensor_num, float *reading)
```
Comment (GoldenBug) [inline common/dev/peci.c:159]: check reading != NULL

File: common/dev/peci.c [hunk 14]
```diff
@@ -0,0 +1,268 @@
...
+		} else if (sensor_num == SENSOR_NUM_TEMP_CPU_TJMAX) {
+			val = read_buf[3];
+		} else {
+			val = read_buf[1];
+		}
+
+		*reading = (float)val;
+		cfg->cache = *reading;
+		cfg->cache_status = SENSOR_READ_SUCCESS;
+		free(read_buf);
```
Comment (GoldenBug) [inline common/dev/peci.c:264]: SAFE_FREE

File: common/dev/tmp75.c [hunk 15]
```diff
@@ -0,0 +1,41 @@
+#include "tmp75.h"
+
+#include <stdio.h>
+
+#include "sensor.h"
+#include "hal_i2c.h"
+
+#define RETRY 5
+
+bool tmp75_read(uint8_t sensor_num, float *reading)
```
Comment (GoldenBug) [inline common/dev/tmp75.c:10]: Check reading != NULL

File: common/lib/libutil.c [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+				uint8_t *data)
+{
+	ipmi_msg ipmi_message;
+	ipmi_message.seq_source = seq_source;
+	ipmi_message.netfn = netFn;
+	ipmi_message.cmd = command;
+	ipmi_message.InF_source = source_inft;
+	ipmi_message.InF_target = target_inft;
+	ipmi_message.data_len = data_len;
+	if (data_len != 0) {
```
Comment (GoldenBug) [inline common/lib/libutil.c:29]: && data != NULL ?

**Review by GoldenBug — CHANGES_REQUESTED** [review]

File: common/dev/adc.c (hunk 1 above)
Comment (BonnieLo) [reply common/dev/adc.c:79]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223 .

File: common/lib/libutil.c (hunk 16 above)
Comment (BonnieLo) [reply common/lib/libutil.c]: We have fixed this and do the forced update.
Please review it. Thanks.

File: common/dev/adc.c (hunk 2 above)
Comment (BonnieLo) [reply common/dev/adc.c:122]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/adm1278.c (hunk 3 above)
Comment (BonnieLo) [reply common/dev/adm1278.c:76]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/guid.c (hunk 4 above)
Comment (BonnieLo) [reply common/dev/guid.c:29]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/isl69260.c (hunk 5 above)
Comment (BonnieLo) [reply common/dev/isl69260.c:13]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/isl69260.c (hunk 6 above)
Comment (BonnieLo) [reply common/dev/isl69260.c:32]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/nvme.c (hunk 7 above)
Comment (BonnieLo) [reply common/dev/nvme.c:74]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/pch.c (hunk 8 above)
Comment (BonnieLo) [reply common/dev/pch.c:40]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/peci.c (hunk 9 above)
Comment (BonnieLo) [reply common/dev/peci.c:69]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/adc.c (hunk 1 above)
Comment (BonnieLo) [reply common/dev/adc.c:79]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/adm1278.c (hunk 3 above)
Comment (BonnieLo) [reply common/dev/adm1278.c:76]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/guid.c (hunk 4 above)
Comment (BonnieLo) [reply common/dev/guid.c:29]: Hi @GoldenBug 
The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/isl69260.c (hunk 5 above)
Comment (BonnieLo) [reply common/dev/isl69260.c:13]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/peci.c (hunk 10 above)
Comment (BonnieLo) [reply common/dev/peci.c:15]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/peci.c (hunk 11 above)
Comment (BonnieLo) [reply common/dev/peci.c:44]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/peci.c (hunk 12 above)
Comment (BonnieLo) [reply common/dev/peci.c:154]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/peci.c (hunk 13 above)
Comment (BonnieLo) [reply common/dev/peci.c:159]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/peci.c (hunk 14 above)
Comment (BonnieLo) [reply common/dev/peci.c:264]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/dev/tmp75.c (hunk 15 above)
Comment (BonnieLo) [reply common/dev/tmp75.c:10]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

File: common/lib/libutil.c (hunk 16 above)
Comment (BonnieLo) [reply common/lib/libutil.c]: We have fixed this and do the forced update.
Please review it. Thanks.

File: common/dev/adc.c (hunk 2 above)
Comment (BonnieLo) [reply common/dev/adc.c:122]: The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

**Comment by DelphineChiu** [comment]
Dependency: #214

File: common/hal/hal_gpio.c [hunk 17]
```diff
@@ -50,13 +50,13 @@ void irq_callback(const struct device *dev, struct gpio_callback *cb, uint32_t p
...
 	}
 	if (index == GPIO_GROUP_SIZE) {
-		printk("irq_callback: pin %x not found\n", pins);
+		printf("irq_callback: pin %x not found\n", pins);
 		return;
 	}
 	gpio_num = (group * GPIO_GROUP_SIZE) + pins;
 
 	if (gpio_cfg[gpio_num].int_cb == NULL) {
-		printk("CB func pointer NULL for gpio num %d\n", gpio_num);
```
Comment (GoldenBug) [inline common/hal/hal_gpio.c:59]: CB -> Callback

File: common/lib/util_sys.c [hunk 18]
```diff
@@ -83,113 +75,124 @@ __weak int pal_submit_12v_cycle_slot()
...
+void set_me_firmware_mode(uint8_t me_fw_mode)
 {
-	ipmi_msg *me_msg;
-	ipmb_error status;
-
-	me_msg = (ipmi_msg *)malloc(sizeof(ipmi_msg));
-	if (me_msg == NULL) {
-		printf("[%s] Failed to allocate memory\n", __func__);
+	if ((me_fw_mode != ME_FW_RECOVERY) && (me_fw_mode != ME_FW_RESTORE)) {
+		printf("Not support the ME frimware mode 0x%x setting", me_fw_mode);
```
Comment (GoldenBug) [inline common/lib/util_sys.c:95]: "Unsupported ME firmware mode 0x%x setting"

File: common/lib/util_sys.c [hunk 19]
```diff
@@ -83,113 +75,124 @@ __weak int pal_submit_12v_cycle_slot()
...
+		data[0] = INTEL_IANA & 0xFF;
+		data[1] = (INTEL_IANA >> 8) & 0xFF;
+		data[2] = (INTEL_IANA >> 16) & 0xFF;
+		data[3] = me_fw_mode;
+		*me_msg = construct_ipmi_message(seq_source, NETFN_NM_REQ,
+						 CMD_OEM_NM_FORCE_ME_RECOVERY, SELF, ME_IPMB,
+						 data_len, data);
+		ret = ipmb_read(me_msg, IPMB_inf_index_map[me_msg->InF_target]);
+		if (ret != IPMB_ERROR_SUCCESS) {
+			printf("Failed to set ME frimware mode to 0x%x, ret: 0x%x\n", me_fw_mode,
```
Comment (GoldenBug) [inline common/lib/util_sys.c:122]: ME firmware

File: common/lib/util_sys.c [hunk 20]
```diff
@@ -83,113 +75,124 @@ __weak int pal_submit_12v_cycle_slot()
...
-			printk("ME restore get selftest fail with status: %x\n", status);
-		}
-		if (me_msg != NULL) {
-			free(me_msg);
+			printf("Failed to get ME self test result after setting 0x%x mode, ret: 0x%x\n",
+			       me_fw_mode, ret);
 		}
 	}
+	if (i == retry) {
+		printf("Failed to set ME frimware mode to 0x%x, retry time: %d\n", me_fw_mode,
```
Comment (GoldenBug) [inline common/lib/util_sys.c:169]: ME firmware

File: meta-facebook/yv35-cl/src/main.c [hunk 21]
```diff
@@ -42,8 +48,7 @@ void set_sys_status()
 
 void main(void)
 {
-	uint8_t proj_stage = (FIRMWARE_REVISION_1 & 0xf0) >> 4;
```
Comment (GoldenBug) [inline meta-facebook/yv35-cl/src/main.c:45]: welcome

**Review by GoldenBug — CHANGES_REQUESTED** [review]

File: meta-facebook/yv35-cl/src/main.c [hunk 22]
```diff
@@ -42,8 +48,7 @@ void set_sys_status()
...
-	printk("Hello, wellcome to yv35 craterlake %x%x.%x.%x\n", BIC_FW_YEAR_MSB, BIC_FW_YEAR_LSB,
+	printf("Hello, wellcome to yv35 craterlake %x%x.%x.%x\n", BIC_FW_YEAR_MSB, BIC_FW_YEAR_LSB,
```
Comment (GoldenBug) [inline meta-facebook/yv35-cl/src/main.c:51]: welcome

File: common/lib/util_sys.c (hunk 18 above)
Comment (BonnieLo) [reply common/lib/util_sys.c]: We have fixed this and done the forced update.
Please review it. Thanks.

File: common/lib/util_sys.c (hunk 19 above)
Comment (BonnieLo) [reply common/lib/util_sys.c]: We have fixed this and done the forced update.
Please review it. Thanks.

File: common/lib/util_sys.c (hunk 20 above)
Comment (BonnieLo) [reply common/lib/util_sys.c]: We have fixed this and done the forced update.
Please review it. Thanks.

File: meta-facebook/yv35-cl/src/main.c (hunk 22 above)
Comment (BonnieLo) [reply meta-facebook/yv35-cl/src/main.c]: We have fixed this and done the forced update.
Please review it. Thanks.

**Pull request closed without merging**